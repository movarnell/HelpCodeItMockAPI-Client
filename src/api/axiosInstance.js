// src/api/axiosInstance.js
import axios from 'axios';

const axiosInstance = axios.create({
  baseURL: 'http://localhost:5100', // Adjust if necessary
});

axiosInstance.interceptors.request.use(
    (config) => {
      const token = localStorage.getItem('token');
      console.log('Token from localStorage:', token);
      if (token) {
        config.headers['Authorization'] = `Bearer ${token}`;
      } else {
        console.warn('No token found in localStorage');
      }
      return config;
    },
    (error) => {
      return Promise.reject(error);
    }
  );

export default axiosInstance;