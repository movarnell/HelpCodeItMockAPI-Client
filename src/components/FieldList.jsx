import React, { useEffect, useState } from 'react';
import axiosInstance from '../api/axiosInstance';
import { useParams, Link } from 'react-router-dom';
import Navigation from './Navigation';
import { IoArrowBack } from 'react-icons/io5';

function FieldList() {
  const { id } = useParams(); // Endpoint ID
  const [fields, setFields] = useState([]);

  useEffect(() => {
    const fetchFields = async () => {
      try {
        const response = await axiosInstance.get(`/endpoints/${id}/fields`);
        setFields(response.data);
      } catch (err) {
        console.error(err);
      }
    };

    fetchFields();
  }, [id]);

  return (
    <div className="min-h-screen bg-gray-100">
      <Navigation />
      <div className="py-6 mx-auto max-w-7xl sm:px-6 lg:px-8">
        <div className="flex items-center mb-6">
          <Link
            to="/endpoints"
            className="mr-4 text-indigo-600 transition-colors duration-200 hover:text-indigo-800"
          >
            <IoArrowBack className="w-6 h-6" />
          </Link>
          <h1 className="text-3xl font-bold text-gray-900">Field List</h1>
        </div>
        <div className="mb-6">
          <Link
            to={`/endpoints/${id}/fields/add`}
            className="inline-flex items-center px-4 py-2 text-sm font-medium text-white bg-indigo-600 border border-transparent rounded-md hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
          >
            Add Field
          </Link>
        </div>
        <div className="overflow-hidden bg-white shadow sm:rounded-lg">
          <table className="min-w-full divide-y divide-gray-200">
            <thead className="bg-gray-50">
              <tr>
                <th scope="col" className="px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-500 uppercase">Field Name</th>
                <th scope="col" className="px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-500 uppercase">Data Type</th>
                <th scope="col" className="px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-500 uppercase">Is Required</th>
              </tr>
            </thead>
            <tbody className="bg-white divide-y divide-gray-200">
              {fields.map((field) => (
                <tr key={field.field_id}>
                  <td className="px-6 py-4 text-sm font-medium text-gray-900 whitespace-nowrap">{field.field_name}</td>
                  <td className="px-6 py-4 text-sm text-gray-500 whitespace-nowrap">{field.data_type}</td>
                  <td className="px-6 py-4 text-sm text-gray-500 whitespace-nowrap">{field.is_required ? 'Yes' : 'No'}</td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>
        <div className="mt-8">
          <Link to="/dashboard" className="inline-flex items-center px-4 py-2 text-sm font-medium text-white bg-green-600 border border-transparent rounded-md hover:bg-green-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-green-500">
            Back to Dashboard
          </Link>
        </div>
      </div>
    </div>
  );
}

export default FieldList;