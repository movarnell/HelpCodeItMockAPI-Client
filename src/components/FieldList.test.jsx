import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import FieldList from './FieldList';
import axiosInstance from '../api/axiosInstance';
import { act } from 'react';

jest.mock('../api/axiosInstance');
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
  useParams: () => ({ id: '1' }),
}));

describe('FieldList Component', () => {
  const mockFields = [
    { field_id: 1, field_name: 'name', data_type: 'STRING', is_required: true },
    { field_id: 2, field_name: 'age', data_type: 'INT', is_required: false },
  ];

  const mockNavigate = jest.fn();

  beforeEach(() => {
    axiosInstance.get.mockResolvedValue({ data: mockFields });
    require('react-router-dom').useNavigate.mockReturnValue(mockNavigate);
    mockNavigate.mockClear();
  });

  const renderComponent = async () => {
    await act(async () => {
      render(
        <MemoryRouter initialEntries={['/endpoints/1/fields']}>
          <Routes>
            <Route path="/endpoints/:id/fields" element={<FieldList />} />
          </Routes>
        </MemoryRouter>
      );
    });
    await waitFor(() => expect(axiosInstance.get).toHaveBeenCalled());
  };

  test('renders FieldList component', async () => {
    await renderComponent();
    expect(screen.getByText('Field List')).toBeInTheDocument();
    expect(screen.getByText('name')).toBeInTheDocument();
    expect(screen.getByText('age')).toBeInTheDocument();
  });

  test('navigates to add field page when Add Field button is clicked', async () => {
    await renderComponent();
    fireEvent.click(screen.getByText(/add field/i));
    expect(mockNavigate).toHaveBeenCalledWith('/endpoints/1/fields/add');
  });

  test('navigates back to endpoints when back arrow is clicked', async () => {
    await renderComponent();
    fireEvent.click(screen.getByTestId('back-arrow'));
    expect(mockNavigate).toHaveBeenCalledWith('/endpoints');
  });
});