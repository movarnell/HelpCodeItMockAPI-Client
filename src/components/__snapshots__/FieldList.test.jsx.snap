// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldList Component matches snapshot 1`] = `
<div
  className="min-h-screen bg-gray-100"
>
  <div
    data-testid="navigation"
  />
  <div
    className="py-6 mx-auto max-w-7xl sm:px-6 lg:px-8"
  >
    <div
      className="flex items-center mb-6"
    >
      <a
        className="mr-4 text-indigo-600 transition-colors duration-200 hover:text-indigo-800"
        data-testid="back-arrow"
        href="/endpoints"
      >
        <svg
          className="w-6 h-6"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M244 400 100 256l144-144M120 256h292"
            fill="none"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="48"
          />
        </svg>
      </a>
      <h1
        className="text-3xl font-bold text-gray-900"
      >
        Field List
      </h1>
    </div>
    <div
      className="mb-6"
    >
      <a
        className="inline-flex items-center px-4 py-2 text-sm font-medium text-white bg-indigo-600 border border-transparent rounded-md hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
        href="/endpoints/1/fields/add"
      >
        Add Field
      </a>
    </div>
    <div
      className="overflow-hidden bg-white shadow sm:rounded-lg"
    >
      <table
        className="min-w-full divide-y divide-gray-200"
      >
        <thead
          className="bg-gray-50"
        >
          <tr>
            <th
              className="px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-500 uppercase"
              scope="col"
            >
              Field Name
            </th>
            <th
              className="px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-500 uppercase"
              scope="col"
            >
              Data Type
            </th>
            <th
              className="px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-500 uppercase"
              scope="col"
            >
              Is Required
            </th>
            <th
              className="px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-500 uppercase"
              scope="col"
            >
              Actions
            </th>
             
          </tr>
        </thead>
        <tbody
          className="bg-white divide-y divide-gray-200"
        />
      </table>
    </div>
    <div
      className="mt-8"
    >
      <a
        className="inline-flex items-center px-4 py-2 text-sm font-medium text-white bg-green-600 border border-transparent rounded-md hover:bg-green-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-green-500"
        href="/dashboard"
      >
        Back to Dashboard
      </a>
    </div>
  </div>
</div>
`;
